apps/common/models.py

```diff
@@ -63,3 +72,30 @@ class ArchivableResource(Model):
     class Meta(TypedModelMeta):  # type: ignore[reportIncompatibleVariableOverride]
         abstract = True
         ordering = ["-id"]
+
+
+class FirebaseResource(Model):
+    firebase_push_status: int | None = IntegerChoicesField(  # type: ignore[reportAssignmentType]
+        choices_enum=FirebasePushStatusEnum,
+        null=True,
+        blank=True,
+    )
+    firebase_last_pushed = models.DateTimeField(
+        null=True,
+        blank=True,
+        help_text=gettext_lazy("The latest time when project was pushed to firebase"),
+    )
+
+    @property
+    def firebase_push_status_enum(self):
+        if self.firebase_push_status:
+            return FirebasePushStatusEnum(self.firebase_push_status)
+        return None
+
+    def update_firebase_push_status(self, firebase_push_status: FirebasePushStatusEnum, *, commit: bool = True):
+        self.firebase_push_status = firebase_push_status
+        if commit:
+            self.save(update_fields=("firebase_push_status",))
+
+    class Meta(TypedModelMeta):  # type: ignore[reportIncompatibleVariableOverride]
+        abstract = True
```

Comment:
Let's use this in project and remove the usage in the project before we merge this